R/expr__string.R

```diff
@@ -81,93 +105,78 @@ ExprStr_strptime = function(
-#' conversion.
-#'
-#' @return Expr
-#'
-#'
+#' @inheritParams ExprStr_strptime
+#' @format Format to use for conversion. Refer to
+#' [the chrono crate documentation](https://docs.rs/chrono/latest/chrono/format/strftime/index.html)
+#' for the full specification. Example: `"%H:%M:%S"`.
+#' If `NULL` (default), the format is inferred from the data.
+#' @return [Expr][Expr_class] of Time type
+#' @seealso
+#' - [`<Expr>$str$strptime()`][ExprStr_strptime]
 #' @examples
-#' pl$DataFrame(str_time = c("01:20:01", "28:00:02", "03:00:02"))$
-#'   with_columns(time = pl$col("str_time")$str$to_time(strict = FALSE))
+#' s = pl$Series(c("01:00", "02:00", "03:00"))
+#'
+#' s$str$to_time("%H:%M")
 ExprStr_to_time = function(format = NULL, strict = TRUE, cache = TRUE) {
   .pr$Expr$str_to_time(self, format, strict, cache) |>
     unwrap("in $str$to_time():")
 }
 
+# TODO for 0.16.0: should not allow positional arguments except for the first one
 #' Convert a String column into a Datetime column
 #'
-#' @param format Format to use for conversion. See `?strptime` for possible
-#' values. Example: "%Y-%m-%d %H:%M:%S". If `NULL` (default), the format is
-#' inferred from the data. Notice that time zone `%Z` is not supported and will
-#' just ignore timezones. Numeric time zones like `%z` or `%:z`  are supported.
-#' @param time_unit String (`"ns"`, `"us"`, `"ms"`) or integer.
-#' @param time_zone String describing a timezone. If `NULL` (default), `"GMT` is
-#' used.
-#' @param strict If `TRUE` (default), raise an error if a single string cannot
-#' be parsed. If `FALSE`, parsing failure will produce a polars `null`.
-#' @param exact If `TRUE` (default), require an exact format match. Otherwise,
-#' allow the format to match anywhere in the target string.
-#' @param cache Use a cache of unique, converted dates to apply the datetime
-#' conversion.
-#' @param ambiguous Determine how to deal with ambiguous datetimes:
-#' * `"raise"` (default): raise
-#' * `"earliest"`: use the earliest datetime
-#' * `"latest"`: use the latest datetime
-#'
-#' @return Expr
-#'
-#'
-#' @examples
-#' pl$DataFrame(str_date = c("2009-01-02 01:00", "2009-01-03 02:00", "2009-1-4 3:00"))$
-#'   with_columns(datetime = pl$col("str_date")$str$to_datetime(strict = FALSE))
+#' @inheritParams ExprStr_strptime
+#' @param time_unit Unit of time for the resulting Datetime column. If `NULL` (default),
+#' the time unit is inferred from the format string if given,
+#' eg: `"%F %T%.3f"` => [`pl$Datetime("ms")`][pl_Datetime].
+#' If no fractional second component is found, the default is `"us"` (microsecond).
+#' @param time_zone for the resulting [Datetime][pl_Datetime] column.
+#' @param exact If `TRUE` (default), require an exact format match. If `FALSE`, allow the format to match
+#' anywhere in the target string. Note that using `exact = FALSE` introduces a performance
+#' penalty - cleaning your data beforehand will almost certainly be more performant.
```

Comment:
Aren't those already inherited from `$strptime()`?

Reply:
I recall, because of the reference to the Time type, it cannot be reused.